### PR TITLE
cmake: add additional option -Wdeclaration-after-statement support

### PR DESCRIPTION
**Pull request by lxbsz**
Warn about C99 declaration after statement to make the code tidy and
more readable.

Signed-off-by: Xiubo Li <lixiubo@cmss.chinamobile.com>

### CODE REVIEW

**Comment by lxbsz** [comment]
@pkalever 
please help review. thanks.

File: api.c [hunk 1]
```diff
@@ -442,10 +442,8 @@ int tcmu_emulate_evpd_inquiry(
 		 * WWN, but this is what the kernel does, and it's nice for our
 		 * values to match.
 		 */
-		char *p = wwn;
-		bool next = true;
-		int i = 7;
-		for ( ; *p && i < 20; p++) {
+		next = true;
+		for (p = wwn, i = 7; *p && i < 20; p++) {
```
Comment (pkalever) [inline api.c]: i = 7;
You have already initialized 'i' with 7 :-)

**Comment by mikechristie** [comment]
Looks ok to me. Just handle pkalever's comment.

File: api.c (hunk 1 above)
Comment (lxbsz) [inline api.c]: Oh Yes, good catch and thanks very much.

**Comment by lxbsz** [comment]
Already updated, thanks @pkalever

**Pull request merged**